### PR TITLE
RNG reloaded

### PR DESCRIPTION
**Pull request by matbesancon**
This PR standardizes all random functions which now take a `rng` argument or keyword argument.

Still failing on one inference, no idea why

### CODE REVIEW

**Comment by sbromberger** [comment]
Can you base this against `2.0.0-dev` instead, please?

File: src/Parallel/dominatingset/minimal_dom_set.jl [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 
 ### Optional Arguements
 - `parallel=:threads`: If `parallel=:distributed` then the multiprocessor implementation is
 used. This implementation is more efficient if `reps` is large.
 - If `seed >= 0`, a random generator of each process/thread is seeded with this value.
 """
-dominating_set(g::AbstractGraph{T}, reps::Integer, alg::MinimalDominatingSet; parallel=:threads, seed=-1) where T <: Integer = 
-LightGraphs.Parallel.generate_reduce(g, (g::AbstractGraph{T})->LightGraphs.dominating_set(g, alg; seed=seed), 
-(x::Vector{T}, y::Vector{T})->length(x)<length(y), reps; parallel=parallel)
+function dominating_set(g::AbstractGraph{T}, reps::Integer, alg::MinimalDominatingSet; parallel=:threads, seed=-1) where T <: Integer
```
Comment (sbromberger) [inline src/Parallel/dominatingset/minimal_dom_set.jl]: Should this take an RNG instead of a seed?

Comment (matbesancon) [reply src/Parallel/dominatingset/minimal_dom_set.jl]: good catch, however, we can't provide a rng directly for race conditions. So I suggest providing a function that gives a rng, like `() -> MersenneTwister(1)`

**Comment by sbromberger** [comment]
closed via #1328.

**Pull request closed without merging**